notebooks/prepare_data.ipynb

```diff
@@ -0,0 +1,187 @@
+    "    plt.close(fig)\n",
+    "\n",
+    "    duration = get_duration(agg_read_data)\n",
+    "\n",
+    "    return agg_read_data, duration"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "outputs": [
+    {
+     "data": {
+      "text/plain": "['/11_experiments_data_2021_05_05_21_37',\n '/12_experiments_data_2021_05_05_21_37',\n '/13_experiments_data_2021_05_05_21_37',\n '/20_experiments_data_2021_05_05_21_37']"
+     },
+     "execution_count": 26,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "nodes_directories = get_subdirectories()\n",
+    "nodes_directories"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "outputs": [
+    {
+     "data": {
+      "text/plain": "['/11_experiments_data_2021_05_05_21_37/countDistinctTicketNumber',\n '/12_experiments_data_2021_05_05_21_37/countDistinctTicketNumber',\n '/13_experiments_data_2021_05_05_21_37/countDistinctTicketNumber',\n '/20_experiments_data_2021_05_05_21_37/countDistinctTicketNumber']"
+     },
+     "execution_count": 27,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "data_directories = []\n",
+    "for directory in nodes_directories:\n",
+    "    cur_node_subdirectories = get_subdirectories(directory)\n",
+    "    data_directories.append(cur_node_subdirectories)\n",
+    "\n",
+    "data_directories = [item for sublist in data_directories for item in sublist]\n",
+    "data_directories"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "outputs": [],
+   "source": [
+    "for experiment_directory in data_directories:\n",
+    "    path = \"./../experiments_data\" + experiment_directory\n",
+    "    p = Path(path)\n",
+    "    iteration = 1\n",
+    "    agg_base_data = pd.DataFrame()\n",
+    "    experiments_durations = []\n",
+    "\n",
+    "    for file in p.glob('*.csv'):\n",
+    "        if iteration == 1:\n",
+    "            agg_base_data, duration = preprocess_file(path, file.name, iteration)\n",
+    "            experiments_durations.append(duration)\n",
+    "\n",
+    "        else:\n",
+    "            agg_new_data, duration = preprocess_file(path, file.name, iteration)\n",
+    "            experiments_durations.append(duration)\n",
+    "            \n",
+    "            agg_base_data = pd.concat((agg_base_data, agg_new_data))\n",
+    "\n",
+    "        iteration += 1\n",
+    "\n",
+    "    experiment_duration_mean = mean(experiments_durations)\n",
+    "\n",
+    "    by_rows_index = agg_base_data.groupby(agg_base_data.index)\n",
```

Comment:
do you really need another variable here? Maybe try to invoke `groupby` and `mean` one by one